src/mlir/error.h

```diff
@@ -19,49 +19,78 @@
 // They all must have a description. Parsing related errors must also have a
 // file location, pointing to the original place the failing function is
 // trying to work with. Stacking multiple errors generate a stack trace.
+//
+// FIXME: The descriptions are initialised by the caller directly with s
+// string value, which doesn't work when the error could be emitted in
+// multiple languages. We need a mechanism where strings could come from
+// any language with a formatted output for a list of arguments, filled in
+// by the caller.
 namespace mlir::verona
 {
   // RuntimeError: Errors that have no relation to the source file
   class RuntimeError : public llvm::ErrorInfo<RuntimeError>
   {
+    // Description of the error
     std::string desc;
 
   public:
+    // Internal ID to control matching
     static char ID;
     RuntimeError(llvm::StringRef desc) : desc(desc) {}
+
+    // Public ErrorInfo interface, requires raw_ostream to output
+    // the string in the right way. This is used by error handlers.
     void log(llvm::raw_ostream& OS) const override
     {
       OS << desc;
     }
+    // Public ErrorInfo interface, converts to error_code for
+    // interoperation with other error handling mechanisms.
```

Comment:
```suggestion
    // interoperation with other LLVM error handling mechanisms.
```

Reply:
Not necessarily. `std::error_code` is standard and can be used by other mechanisms, standard or not.

Reply:
Ah, I see.  Perhaps then:
```suggestion
interoperation with error-handling mechanisms that use the C++11 error types.
```